### PR TITLE
Show detail config in yaml or json.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Show detail config in yaml or json.
Hides daprsystem
Hides MTLS
Shows Metrics.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #487 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

```
$ ./dist/darwin_amd64/release/dapr configurations -k 
  NAME       TRACING-ENABLED  METRICS-ENABLED  AGE  CREATED              
  appconfig  true             true             1h   2020-12-10 22:01.59  
(23:09:19)[asouza@MININT-UEMT0H6] ~/.../dapr/cli (configuration-yaml {3} U:4 ✗) 
$ ./dist/darwin_amd64/release/dapr configurations -k -o yaml
name: appconfig
spec:
  httppipelinespec:
    handlers: []
  tracingspec:
    samplingrate: "1"
    zipkin:
      endpointaddresss: ""
  metricspec:
    enabled: true
  mtlsspec:
    enabled: false
    workloadcertttl: ""
    allowedclockskew: ""
  secrets:
    scopes: []
  accesscontrolspec:
    defaultAction: ""
    trustDomain: ""
    policies: []
(23:09:25)[asouza@MININT-UEMT0H6] ~/.../dapr/cli (configuration-yaml {3} U:4 ✗) 
$ ./dist/darwin_amd64/release/dapr configurations -k -o json
{
  "name": "appconfig",
  "spec": {
    "httpPipeline": {
      "handlers": null
    },
    "tracing": {
      "samplingRate": "1",
      "zipkin": {
        "endpointAddress": ""
      }
    },
    "metric": {
      "enabled": true
    },
    "mtls": {
      "enabled": false,
      "workloadCertTTL": "",
      "allowedClockSkew": ""
    },
    "secrets": {
      "scopes": null
    },
    "accessControl": {
      "defaultAction": "",
      "trustDomain": "",
      "policies": null
    }
  }
}
```

### CODE REVIEW

File: pkg/kubernetes/configurations.go [hunk 1]
```diff
@@ -6,49 +6,104 @@
...
-		co = append(co, ConfigurtionsOutput{
-			TracingEnabled:  tracingEnabled(c.Spec.TracingSpec),
-			Name:            c.GetName(),
-			MTLSEnabled:     c.Spec.MTLSSpec.Enabled,
-			WorkloadCertTTL: c.Spec.MTLSSpec.WorkloadCertTTL,
-			ClockSkew:       c.Spec.MTLSSpec.AllowedClockSkew,
-			Created:         c.CreationTimestamp.Format("2006-01-02 15:04.05"),
-			Age:             age.GetAge(c.CreationTimestamp.Time),
+		confName := c.GetName()
+		if (confName != "daprsystem") && (name == "" || confName == name) {
```
Comment (wcs1only) [inline pkg/kubernetes/configurations.go:49]: It took me a few rereadings to parse this logic. Would it be more clear if expressed like this?:
```golang
if (confName == "daprsystem") {
    continue
}

if (name == "" || name == confName) {
...
```

**Review by wcs1only — COMMENTED** [review]
Minor nit. Up to you if you want to change it. Also, can we take this opportunity to improve the test coverage? I will leave it up to your judgement as to wether or not that is too much effort for this change.

**Comment by artursouza** [comment]
> Minor nit. Up to you if you want to change it. Also, can we take this opportunity to improve the test coverage? I will leave it up to your judgement as to wether or not that is too much effort for this change.

Code coverage is bothering me here too. Let me see if I can separate the print part of the code, so I can unit test the rest.

**Comment by msfussell** [comment]
1) Will I only see the name for those that have setting? For example if there is no "accessControl" set, then I will not see this, rather than it being empty ""?
2) In future we could expand the default view if needed (Currently Tracing and Metrics enabled)?
3) The tracing is an interpretation of samplingrate being zero or non-zero setting, correct? If I put samplingrate = 0 does tracing-enabled become false?
4) Age refers to the config file?

**Comment by artursouza** [comment]
> This is great! Much more expressive and informative.
> 
> 1. Will I only see the name for those that have setting? For example if there is no "accessControl" set, then I will not see this, rather than it being empty ""?
We return empty for those that do not have been set yet. This is the K8s Spec for the object.
> 2. In future we could expand the default view if needed (Currently Tracing and Metrics enabled)?
Yes, list view is 100% under our control.
> 3. The tracing is an interpretation of samplingrate being zero or non-zero setting, correct? If I put samplingrate = 0 does tracing-enabled become false?
Yes, it means tracing rate > 0.
> 4. Age refers to the config file?
It is the age of the config in K8s. This cmd does not work in standalone mode.

File: pkg/kubernetes/configurations.go (hunk 1 above)
Comment (artursouza) [reply pkg/kubernetes/configurations.go]: Done.

**Review by wcs1only — APPROVED** [review]
Awesome, thanks for adding those tests.

**Pull request merged**